examples/now_in_dart_flutter/lib/features/detail/dart_detail/data/dart_detail_repository.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:fpdart/fpdart.dart';
+import 'package:now_in_dart_flutter/core/data/data.dart';
+import 'package:now_in_dart_flutter/core/domain/failure.dart';
+import 'package:now_in_dart_flutter/core/domain/fresh.dart';
+import 'package:now_in_dart_flutter/features/detail/core/data/detail_dto.dart';
+import 'package:now_in_dart_flutter/features/detail/core/domain/detail.dart';
+import 'package:now_in_dart_flutter/features/detail/dart_detail/data/dart_detail_local_service.dart';
+import 'package:now_in_dart_flutter/features/detail/dart_detail/data/dart_detail_remote_service.dart';
+
+typedef _DartDetailOrFailure = TaskEither<Failure, Fresh<Detail>>;
+
+class DartDetailRepository {
+  DartDetailRepository({
+    DartDetailLocalService? localService,
+    DartDetailRemoteService? remoteService,
+  })  : _localService = localService ?? DartDetailLocalService(),
+        _remoteService = remoteService ?? DartDetailRemoteService();
+
+  final DartDetailLocalService _localService;
+  final DartDetailRemoteService _remoteService;
+
+  _DartDetailOrFailure getDartDetail(int id) {
+    return TaskEither.Do(
+      (_) async {
+        final remoteResponse = await _(
+          _remoteService.getDartChangelogDetail(id),
+        );
+
+        return remoteResponse.when(
```

Comment:
This returns an `async` response. You should always use `_` when executing async functions inside `Do`

Reply:
I tried but couldn't figure out way to refactor my code. I'd really appreciate your help.